### DISCUSSION TITLE
Minimal local install for development

### DISCUSSION THREAD

**Discussion by liamfallon** [nephio-project/nephio#414] [Development] [answered]
The following minimal local install seems to work for me on a Macbook Pro M1 to get the nephio-operators running in vscode talking to a local kind cluster running in Docker.

1. Install Docker
2. install kind
`brew install kind`
3. Create the management cluster
`kind create cluster -n mgmt`
4. Install gitea
`kpt pkg get https://github.com/nephio-project/nephio-example-packages/tree/main/gitea gitea`
`kpt live init gitea`
`kpt live apply gitea`
5. Make gitea visible on host machine
`kubectl port-forward -n gitea svc/gitea 3000:3000`
6. Install Porch
`kpt pkg get https://github.com/nephio-project/nephio-example-packages/tree/main/porch-dev porch-dev`
`kpt live init porch-dev`
`kpt live apply porch-dev`
7. Install configsync
`kpt pkg get https://github.com/nephio-project/nephio-example-packages/tree/main/configsync configsync`
`kpt live init configsync`
`kpt live apply configsync`
8. Install the resource backend
`kpt pkg get https://github.com/nephio-project/nephio-example-packages/tree/main/resource-backend resource-backend`
`kpt live init resource-backend`
 `kpt live apply resource-backend`
9. Load the Nephio CRDs
`kpt pkg get https://github.com/nephio-project/nephio-example-packages/tree/main/nephio-controllers nephio-controllers`
`ls nephio-controllers/crd/bases/*.yaml | xargs -n1 kubectl apply -f`

Now you can run the Nephio controllers in vscode using the following configuration in your `launch.json` file

```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Launch Nephio Controller Manager",
            "type": "go",
            "request": "launch",
            "mode": "auto",
            "program": "${workspaceFolder}/operators/nephio-controller-manager",
            "args": [
                "--reconcilers",
                "*"
            ],
            "env": {
				"GIT_URL": "http://localhost:3000",
                "GIT_NAMESPACE": "gitea"
			},
        }
    ]
}

**Comment by electrocucaracha** [comment]
Hey @liamfallon , I think the steps 3-8 can be scripted as follows:

```bash
cat <<EOF | kind create cluster --config=-
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
  - role: control-plane
    image: kindest/node:v1.28.0
    extraPortMappings:
      - containerPort: 3000
        hostPort: 3000
EOF
pushd "$(mktemp -d -t "kpt-pkg-XXX")" >/dev/null
for pkg in gitea porch-dev configsync resource-backend; do
    kpt pkg get "https://github.com/nephio-project/nephio-example-packages.git/${pkg}@main" "$pkg"
    kpt live init "$pkg"
    kpt live apply "$pkg"
done
popd >/dev/null
```

BTW, the gitea service type has been [recently changed to](https://github.com/nephio-project/nephio-example-packages/pull/33) `LoadBalancer`, so it's necessary to modify it to `ClusterIP`

```bash
KUBE_EDITOR='sed -i "s|  type\: .*|  type\: ClusterIP|g"' kubectl edit service -n gitea gitea
```

**Reply by liamfallon** [reply]
Thanks Victor, this is very useful.

**Comment by liamfallon** [comment, marked as answer]
I have added this to the Nephio documentation in this PR

https://github.com/nephio-project/docs/pull/69
